### PR TITLE
feat(module:icon): switch to SVG icons

### PR DESCRIPTION
**Pull request by wendzhue**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Icons based on font files.

Issue Number: N/A

## What is the new behavior?

Icons based on SVG. 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

But we should make a break change for this in the future.

## Other information

localhost_49152_components_icon_zh
localhost_49152_components_icon_zh 1

### CODE REVIEW

File: components/icon/demo/basic.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'nz-demo-icon-basic',
+  template: `
+    <i icon [type]="'home'"></i>
```
[the remark this answers is not in the document]
Comment (wendzhue) [reply components/icon/demo/basic.ts]: 嗯, 现在 API 的设计只考虑了和 antd 兼容. 这个 API 的确更好一些.

**Comment by wendzhue** [comment]
Recent CSS update is also included in this pull request for previewing the result.

TODO: Doc needs to be updated, and some components broke because of icons are replaced with SVG icons.

**Comment by wendzhue** [comment]
Waiting for `@ant-design/icons-angular` to release. 🎉

**Comment by wendzhue** [comment]
Netlify is still not OK. 😢

File: components/icon/doc/index.zh-CN.md [hunk 2]
```diff
@@ -3,48 +3,135 @@ category: Components
...
+
+你可以通过 `NzIconService` 的 `changeAssetsSource()` 方法来修改图标资源的位置，这样你就可以部署这些资源到 cdn 上。你的参数会被直接添加到 `assets/` 的前面。
+
+### 双色图标主色
+
+对于双色图标，可以通过修改 `this.iconService.twoToneColor = { primaryColor: '#1890ff' }` 来全局设置图标主色。你还可以访问这个属性获取当前的全局主色。
+
+### 自定义 font 图标
+
+在 `1.7.0` 里，我们提供了一个 `fetchFromIconfont` 方法，方便开发者调用在 iconfont.cn 上自行管理的图标。
```
Comment (vthinkxie) [inline components/icon/doc/index.zh-CN.md]: 从 1.7.0 版本开始

File: components/icon/doc/index.zh-CN.md [hunk 3]
```diff
@@ -3,48 +3,135 @@ category: Components
...
+| `addIcon()` | 用于静态引入图标，可传入多个值（或者用数组解构赋值） | `IconDefinition` |
+| `fetchFromIconfont()` | 用于从 FontIcon 获取图标资源文件 | `NzIconfontOption` |
+| `changeAssetsSource()` | 用于修改动态加载 icon 的资源前缀，使得你可以部署图标资源到你想要的任何位置，例如 cdn | `string` |
 
-- 实心和描线图标保持同名，用 `-o` 来区分，比如 `question-circle`（实心） 和 `question-circle-o`（描线）；
-- 命名顺序：`[图标名]-[形状?]-[描线?]-[方向?]`。
+### Svg 图标
 
-> `?` 为可选。
+在 `1.7.0` 版本后，我们与 Ant Design `3.9.x` 同步，使用了 svg 图标替换了原先的 font 图标，从而带来了以下优势：
```
Comment (vthinkxie) [inline components/icon/doc/index.zh-CN.md]: 从 1.7.0 版本开始

File: components/icon/doc/index.zh-CN.md [hunk 4]
```diff
@@ -3,48 +3,135 @@ category: Components
...
+    {
+      "glob": "**/*",
+      "input": "./node_modules/@ant-design/icons/inline-svg/",
+      "output": "/assets/"
+    }
+  ]
+}
+```
+
+你可以通过 `NzIconService` 的 `changeAssetsSource()` 方法来修改图标资源的位置，这样你就可以部署这些资源到 cdn 上。你的参数会被直接添加到 `assets/` 的前面。
```
Comment (vthinkxie) [inline components/icon/doc/index.zh-CN.md]: 详细说明

File: components/icon/doc/index.zh-CN.md [hunk 5]
```diff
@@ -3,48 +3,135 @@ category: Components
...
-我们为每个图标赋予了语义化的命名，命名规则如下:
+| 方法/属性 | 说明 | 参数 |
+| -------- | ----------- | ---- |
+| `addIcon()` | 用于静态引入图标，可传入多个值（或者用数组解构赋值） | `IconDefinition` |
+| `fetchFromIconfont()` | 用于从 FontIcon 获取图标资源文件 | `NzIconfontOption` |
+| `changeAssetsSource()` | 用于修改动态加载 icon 的资源前缀，使得你可以部署图标资源到你想要的任何位置，例如 cdn | `string` |
 
-- 实心和描线图标保持同名，用 `-o` 来区分，比如 `question-circle`（实心） 和 `question-circle-o`（描线）；
-- 命名顺序：`[图标名]-[形状?]-[描线?]-[方向?]`。
+### Svg 图标
```
Comment (vthinkxie) [inline components/icon/doc/index.zh-CN.md]: SVG

**Pull request merged**